lib/hooks/use-loading.ts

```
@@ -0,0 +1,251 @@
+  };
+}
+
+// Multiple loading states manager
+export function useMultipleLoading<T extends string>(
+  keys: readonly T[],
+  options: UseLoadingOptions = {}
+) {
+  const [states, setStates] = useState<Record<T, LoadingState>>(() =>
+    keys.reduce(
+      (acc, key) => ({
+        ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)